src/main/java/ch/njol/skript/classes/data/DefaultOperations.java

```diff
@@ -0,0 +1,103 @@
+			if (Utils.isInteger(left, right) && right.longValue() >= 0)
+				return (long) Math.pow(left.longValue(), right.longValue());
+			return Math.pow(left.doubleValue(), right.doubleValue());
+		});
+		Arithmetics.registerDifference(Number.class, (left, right) -> {
+			if (Utils.isInteger(left, right))
+				return Math.abs(left.longValue() - right.longValue());
+			return Math.abs(left.doubleValue() - right.doubleValue());
+		});
+		Arithmetics.registerDefaultValue(Number.class, () -> 0L);
+
+		// Vector - Vector
+		Arithmetics.registerOperation(Operator.ADDITION, Vector.class, (left, right) -> left.clone().add(right));
+		Arithmetics.registerOperation(Operator.SUBTRACTION, Vector.class, (left, right) -> left.clone().subtract(right));
+		Arithmetics.registerOperation(Operator.MULTIPLICATION, Vector.class, (left, right) -> left.clone().multiply(right));
+		Arithmetics.registerOperation(Operator.DIVISION, Vector.class, (left, right) -> left.clone().divide(right));
+
+		// Vector - Number
+		// Number - Vector
+		Arithmetics.registerOperation(Operator.MULTIPLICATION, Vector.class, Number.class, (left, right) -> left.clone().multiply(right.doubleValue()), (left, right) -> {
+			double number = left.doubleValue();
+			Vector leftVector = new Vector(number, number, number);
+			return leftVector.multiply(right);
+		});
+		Arithmetics.registerOperation(Operator.DIVISION, Vector.class, Number.class, (left, right) -> {
+			double number = right.doubleValue();
+			Vector rightVector = new Vector(number, number, number);
+			return left.clone().divide(rightVector);
+		}, (left, right) -> {
+			double number = left.doubleValue();
+			Vector leftVector = new Vector(number, number, number);
+			return leftVector.divide(right);
+		});
+		Arithmetics.registerDifference(Vector.class,
+			(left, right) -> new Vector(Math.abs(left.getX() - right.getX()), Math.abs(left.getY() - right.getY()), Math.abs(left.getZ() - right.getZ())));
+		Arithmetics.registerDefaultValue(Vector.class, Vector::new);
+
+		// Timespan - Timespan
+		Arithmetics.registerOperation(Operator.ADDITION, Timespan.class, (left, right) -> new Timespan(left.getMilliSeconds() + right.getMilliSeconds()));
+		Arithmetics.registerOperation(Operator.SUBTRACTION, Timespan.class, (left, right) -> new Timespan(Math.max(0, left.getMilliSeconds() - right.getMilliSeconds())));
+		Arithmetics.registerDifference(Timespan.class, (left, right) -> new Timespan(Math.abs(left.getMilliSeconds() - right.getMilliSeconds())));
+		Arithmetics.registerDefaultValue(Timespan.class, Timespan::new);
+
+		// Timespan - Number
+		Arithmetics.registerOperation(Operator.MULTIPLICATION, Timespan.class, Number.class, (left, right) -> new Timespan(left.getMilliSeconds() * right.longValue()));
+		Arithmetics.registerOperation(Operator.DIVISION, Timespan.class, Number.class, (left, right) -> new Timespan(left.getMilliSeconds() / right.longValue()));
```

Comment:
Dividing by 0, NaN,  or by a non-number results in a stacktrace:
```
[18:38:31 ERROR]: #!#! Stack trace:
[18:38:31 ERROR]: #!#! java.lang.ArithmeticException: / by zero
[18:38:31 ERROR]: #!#!     at Skript.jar//ch.njol.skript.classes.data.DefaultOperations.lambda$static$20(DefaultOperations.java:94)
[18:38:31 ERROR]: #!#!     at Skript.jar//ch.njol.skript.expressions.arithmetic.ArithmeticChain.calculate(ArithmeticChain.java:96)
[18:38:31 ERROR]: #!#!     at Skript.jar//ch.njol.skript.expressions.arithmetic.ArithmeticChain.get(ArithmeticChain.java:82)
```

Negative numbers (or infinity) when multiplying and dividing result in illegal argument stacktrace:
```
[18:41:30 ERROR]: #!#! Stack trace:
[18:41:30 ERROR]: #!#! java.lang.IllegalArgumentException: millis must be >= 0
[18:41:30 ERROR]: #!#!     at Skript.jar//ch.njol.skript.util.Timespan.<init>(Timespan.java:142)
[18:41:30 ERROR]: #!#!     at Skript.jar//ch.njol.skript.classes.data.DefaultOperations.lambda$static$20(DefaultOperations.java:94)
[18:41:30 ERROR]: #!#!     at Skript.jar//ch.njol.skript.expressions.arithmetic.ArithmeticChain.calculate(ArithmeticChain.java:96)
```